pkg/apis/kops/v1alpha1/cluster.go

```diff
@@ -251,6 +254,24 @@ type ClusterSpec struct {
 	Assets *Assets `json:"assets,omitempty"`
 }
 
+// FileAssetSpec defines the structure for a file asset
+type FileAssetSpec struct {
+	// Name is a shortened reference to the asset
+	Name string `json:"name,omitempty"`
+	// Path is the location this file should reside
+	Path string `json:"path,omitempty"`
+	// Mode is unix filemode for the file - defaults to 0400
+	Mode string `json:"mode,omitempty"`
+	// Roles is a list of roles the file asset should be applied, defaults to all
+	Roles []InstanceGroupRole `json:"roles,omitempty"`
+	// Content is the contents of the file
+	Content string `json:"content,omitempty"`
+	// Templated indicates the content is templated
+	Templated bool `json:"templated,omitempty"`
```

Comment:
If we do want to allow Templated, we have to be really careful - it ties us to a particular version of the API objects.  So we should be specifying the API version somehow.  But I think I would rather not include Templated at all, and push people to hooks, because our experience of go templating is that they rapidly become unmaintainable.